app/Http/Controllers/GameController.php

```diff
@@ -4,13 +4,17 @@
 
 use App\Models\Game;
 use Illuminate\Http\Request;
+use App\Services\GameService;
 
 class GameController extends Controller
 {
-    public function gameList(?int $id = null)
+    public function __construct( private GameService $gameService)
     {
+    }
+    public function gameList(){
         return view('game-list', [
-            'games'=>$this->getGames(),
+            'games'=>$this->gameService->getGames(),
+            'greeting'=>'Hello World'
         ]);
     }
 
```

Comment:
the data returned in getGames can be returned in the GameService::getGames() method for now.

also getGameById should just call your service method to get the game by id, then return the view. Basically it can look almost identical to the gameList function.

Reply:
I don't think there should there be a "?" in front of int? In the getGamesById function it is just "int $id". Not sure though!